workshops/react_calendar/README.md

```diff
@@ -0,0 +1,670 @@
+
+  const year = today.getFullYear();
+  const month = today.toLocaleString("default", { month: "long" });
+  const day = today.toLocaleString("default", { weekday: "long" });
+  const date = today.getDate();
+
+  return (
+    <div className="container">
+      <div className="main">
+        <p className="month">{month}</p>
+        <p className="day">{day}</p>
+        <p className="date">{date}</p>
+        <p className="year">{year}</p>
+      </div>
+    </div>
+	);
+};
+
+export default CalendarUI;
+```
+</details>
+
+We have successfully built a basic UI using React. Now let's learn how to manipulate states in React. If you saw the demo at the beginning of the workshop, we were able to navigate back and forth dates. This was done using state manipulation. If the state gets changed, the components are re-rendered and thereby show the new state. Let's start implementing this.
+
+#### 2) ChangeDate
+
+In our components file, let's create another component named `ChangeDate.js`. Try to fill out that file with the basic boilerplate code by referring the 5 steps of writing React.
+
+You should have something like this in your `ChangeDate.js` file.
+
+```jsx
+import React from "react";
+import './components.css'
+
+const ChangeDate = () => {
+  return (
+    <div></div>
+  );
+};
+
+
+export default ChangeDate;
+```
+
+Also, give the `<div>` tags a `className` of `buttons`.
+
+Now states are limited to that component itself. So we can't access our `today` state which is in `CalendarUI` component in our `ChangeDate` component.
+
+So we'll need to pass that state in the form of `props` to our `ChangeDate` component. But for that, `ChangeDate` component needs to be the child component of `CalendarUI`. Confused? You'll understand it better once you write the code.
+
+So we will import our `ChangeDate` component inside our `CalendarUI` component and then we'll render it there.
+
+```jsx
+import React, { useState } from "react";
+import "./components.css";
+import ChangeDate from "./ChangeDate.js"; // IMPORT THE COMPONENT
+
+const CalendarUI = () => {
+  const [today, setToday] = useState(new Date());
+
+  const year = today.getFullYear();
+  const month = today.toLocaleString("default", { month: "long" });
+  const day = today.toLocaleString("default", { weekday: "long" });
+  const date = today.getDate();
+
+  return (
+      <div className="container">
+        <div className="main">
+          <p className="month">{month}</p>
+          <p className="day">{day}</p>
+          <p className="date">{date}</p>
+          <p className="year">{year}</p>
+        </div>
+      </div>
+      <ChangeDate /> {/* RENDER THE COMPONENT */}
+  );
+};
+```
+But now, the browser will be badly yelling at you. That is because we are rendering the `<ChangeDate />` outside the parent element. But here, we can't take it in because it will become a flex child (the `container` className is set to `display: flex`) and will be displayed weirdly on the browser.
```

Comment:
Maybe include a screenshot of the error? Not sure what "browser will be badly yelling at you" means exactly here